packages/core/src/App/Containers/Layout/header/dtrader-v2-contract-detail-header.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { observer } from '@deriv/stores';
+import { Text } from '@deriv-com/quill-ui';
+import { LabelPairedArrowLeftSmBoldIcon } from '@deriv/quill-icons';
+import { Localize } from '@deriv/translations';
+import { useHistory } from 'react-router-dom';
+
+const DTraderContractDetailsHeader = observer(() => {
+    const history = useHistory();
+
+    return (
+        <header className='header contract-details-header-v2'>
+            <React.Suspense fallback={<div />}>
+                <LabelPairedArrowLeftSmBoldIcon
+                    height='22px'
+                    width='13px'
+                    className='arrow'
+                    data-testid='arrow'
+                    onClick={() => history.goBack()}
```

Comment:
Please keep production logic by using the below condition that we have in production for this:
```
const { state } = useLocation<TLocationState>();
const is_from_table_row = !isEmptyObject(state) ? state.from_table_row : false; // this can be shortened.
return is_from_table_row ? history.goBack() : routeBackInApp(history);
```
`from_table_row` will be `true` when the user opens Contract Details from Reports page.
It means that when the user presses the arrow-left button to go back, there are 2 cases:
1. if he opened Contract Details from Reports page, 'Back' button should trigger history.goBack(),
2. if he opened Contract Details from DTrader page or else, 'Back' button should trigger routeBackInApp(history).

Reply:
fixed thanks